### ISSUE TITLE
Something changed here? Cannot read properties of undefined 

### ISSUE THREAD

**Issue by dertuerke** [angular/angular#48922]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

Yes

### Description

Hello,
in Angular 14 it was possible to execute the code like this:
``` typescript
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, OnDestroy } from '@angular/core';
import { ConfigService } from '@shared/services/config.service';
import { Subject, takeUntil } from 'rxjs';

@Component({
  selector: 'app-switch-theme',
  templateUrl: './switch-theme.component.html',
  styleUrls: ['./switch-theme.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class SwitchThemeComponent implements OnDestroy {
  public darkMode: boolean = this.configService.getThemeState();

  private readonly ngUnsubscribe: Subject<void> = new Subject();

  constructor(private readonly configService: ConfigService, private readonly cdr: ChangeDetectorRef) {
    this.configService.$darkMode.pipe(takeUntil(this.ngUnsubscribe)).subscribe((res) => {
      this.darkMode = res;
      this.cdr.markForCheck();
    });
  }

  public switchTheme(mode: boolean): void {
    this.configService.$darkMode.next(mode);
  }

  ngOnDestroy(): void {
    this.ngUnsubscribe.next();
    this.ngUnsubscribe.complete();
  }
}
```

``` typescript
@Injectable({
  providedIn: 'root'
})
export class ConfigService {
  public getThemeState(): boolean {
    return localStorage.getItem('darkMode') === 'true' ? true : false || false;
  }
}
```

Now I have updated my project to Angular 15 and get the following error message:
```
ERROR TypeError: Cannot read properties of undefined (reading 'getThemeState')
    at new SwitchThemeComponent (switch-theme.component.ts:12:49)
    at NodeInjectorFactory.SwitchThemeComponent_Factory [as factory] (switch-theme.component.ts:11:34)
    at getNodeInjectable (core.mjs:3485:44)
    at instantiateAllDirectives (core.mjs:11092:27)
    at createDirectivesInstances (core.mjs:10563:5)
    at ɵɵelementStart (core.mjs:13716:9)
    at Module.ɵɵelement (core.mjs:13771:5)
    at HeadComponent_Template (head.component.html:30:7)
    at executeTemplate (core.mjs:10534:9)
    at renderView (core.mjs:10356:13)
```

But if I add a "question mark" to the service call, it works:
``` typescript
public darkMode: boolean = this.configService?.getThemeState();
```
Why is this now an error? I don't get any errors from VSCode or Eslint.

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 15.1.4
Node: 16.18.1
Package Manager: npm 9.4.1
OS: darwin arm64

Angular: 15.1.2
... animations, cdk, common, compiler, compiler-cli, core, forms
... material, platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1501.4
@angular-devkit/build-angular   15.1.4
@angular-devkit/core            15.1.4
@angular-devkit/schematics      15.1.4
@angular/cli                    15.1.4
@schematics/angular             15.1.4
rxjs                            7.8.0
typescript                      4.9.5
```

**Comment by ajafff** [comment]
https://github.com/microsoft/TypeScript/issues/50971

TL;DR: `@angular/cli` tries to infer the correct/best compilation settings for your target browsers (as configured in your .browserslistrc) or uses a default. In your setup it seems to infer `target: ES2022` and `useDefineForClassFields: true`.
This changes the emit of TypeScript and basically the initialization order of parameter properties changes.

**Comment by dertuerke** [comment]
@ajafff i dont have a .browserlistrc anymore and the target i set to `es2022`.
I have not set the option `useDefineForClassFields`.

Is this a bug that can be fixed or do I have to put a "question mark" everywhere now?

**Comment by JoostK** [comment]
Did you update using `ng update`, or only by changing the versions to Angular 15 yourself? There is a migration in Angular 15 that sets `useDefineForClassFields: false`; but if you updated manually then no migrations would have been executed.

**Comment by dertuerke** [comment]
@JoostK aaaaaahhhh i see. i forgot the copy the option. Thx for the help :)
